src/FlattenMsg.chpl

```diff
@@ -13,17 +13,18 @@ module FlattenMsg {
   const fmLogger = new Logger(logLevel);
 
   proc segFlattenMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
-    var (name, objtype, returnSegsStr, regexStr, delimJson) = payload.splitMsgToTuple(5);
-    const returnSegs: bool = returnSegsStr.toLower() == "true";
-    const regex: bool = regexStr.toLower() == "true";
-    const arr = jsonToPdArray(delimJson, 1);
+    var msgArgs = parseMessageArgs(payload, 5);
+    const objtype = msgArgs.getValueOf("objtype");
+    const returnSegs: bool = msgArgs.get("return_segs").getBoolValue();
+    const regex: bool = msgArgs.get("regex").getBoolValue();
+    const arr = msgArgs.get("delim").getList(1);
     const delim: string = arr[arr.domain.low];
```

Comment:
if we can get away with just passing `delimiter`, I think this can be simplified to just be 
```chapel
const delim: string = msgArgs.getValueOf("delim");
```

Reply:
Done